### PR TITLE
Adopt new control scheme in SliceRenderer

### PR DESCRIPTION
**Pull request by sgpearse**
Fix #3035.  

This PR introduces PSliceController, a widget that can be used by the both Slice and Contour renderers to position their planes.

### CODE REVIEW

**Review by StasJ — APPROVED** [review]

**Review by clyne — COMMENTED** [review]
The Normal mode seems to be a no-op. Or possibly it becomes a no-op if you first use rotation mode and then try to use Normal mode.

**Comment by sgpearse** [comment]
@clyne - On it.  Thanks for the catch.

**Review by clyne — COMMENTED** [review]
The UI looks great. Just once question about the implementation

File: lib/render/SliceRenderer.cpp [hunk 1]
```diff
@@ -186,9 +199,17 @@ int SliceRenderer::_regenerateSlice()
...
-    pd.normal = {normal[0], normal[1], normal[2]};
+
+    if (_cacheParams.sliceOrientationMode == (int)RenderParams::SlicePlaneOrientationMode::Normal)
+        pd.normal = _cacheParams.sliceNormal;
+    else
+        pd.normal = ToDoubleVec(ArbitrarilyOrientedRegularGrid::GetNormalFromRotations(_cacheParams.sliceRotation));
+
+    glm::vec3 o = {_cacheParams.xOrigin, _cacheParams.yOrigin, _cacheParams.zOrigin};
+    glm::vec3 n = {pd.normal[0], pd.normal[1], pd.normal[2]};
+    glm::vec3 offsetOrigin = o + n * (float)_cacheParams.sliceOffset;
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply lib/render/SliceRenderer.cpp:210]: No, it's not guaranteed at this point in the code.  This is handled in [ArbitrarilyOrientedRegularGrid.cpp:30](https://github.com/NCAR/VAPOR/blob/69ae174a92ee5cda5052c646fc1443b9937434cc/lib/vdc/ArbitrarilyOrientedRegularGrid.cpp#L30).

**Pull request merged**